src/app/collection/pages/security-injections/security-injections.component.ts

```diff
@@ -1,15 +1,50 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, OnInit, OnDestroy, HostListener } from '@angular/core';
+import { Router } from '@angular/router';
+import { LearningObject } from '@entity';
+import { CollectionService } from 'app/core/collection.service';
+import { FeaturedObjectsService } from 'app/core/featuredObjects.service';
+import { NavbarService } from '../../../core/navbar.service';
 
 @Component({
   selector: 'clark-security-injections',
   templateUrl: './security-injections.component.html',
-  styleUrls: ['./security-injections.component.scss']
+  styleUrls: ['./security-injections.component.scss'],
 })
-export class SecurityInjectionsComponent implements OnInit {
+export class SecurityInjectionsComponent implements OnInit, OnDestroy {
 
-  constructor() { }
+  abvCollection = 'secinj';
+  collection;
+  learningObjects: LearningObject[];
+  isDesktop: boolean;
 
-  ngOnInit(): void {
+  constructor(
+    private navbarService: NavbarService,
+    private collectionService: CollectionService,
+    private featureService: FeaturedObjectsService,
+    private router: Router,
+    ) {
+      this.isDesktop = window.innerWidth >= 768;
+    }
+
+    @HostListener('window:resize', ['$event'])
+    onResize(event: Event) {
+    this.isDesktop = window.innerWidth >= 768;
+    }
+
+  async ngOnInit() {
+
+    this.navbarService.show();
+
+    this.collection = await this.collectionService.getCollection(this.abvCollection);
+
+    this.collection = await this.collectionService.getCollectionMetadata(this.abvCollection).catch(e => {
+      if (e.status === 404) {
+        this.router.navigate(['not-found']);
+      }
+    });
```

Comment:
`this.collection` is being assigned to the return value from the second assignment line 40. Is the first assignment on line 38 necessary?

Reply:
Good catch, that was an accidental duplicate